Sources/Fullscreen/SettingsDetailsView/SettingsDetailsView.swift

```diff
@@ -0,0 +1,222 @@
+//
+//  Copyright © 2019 FINN AS. All rights reserved.
+//
+
+import UIKit
+
+// MARK: - Model
+public protocol SettingsDetailsViewModel {
+    var icon: UIImage { get }
+    var title: String { get }
+    var primaryButtonStyle: Button.Style { get }
+    var primaryButtonTitle: String { get }
+    func text(for state: SettingsDetailsView.State) -> String
+    func textAlignment(for state: SettingsDetailsView.State) -> NSTextAlignment
+    func secondaryButtonTitle(for state: SettingsDetailsView.State) -> String?
+}
+
+// MARK: - Delegate
+public protocol SettingsDetailsViewDelegate: AnyObject {
+    func settingsDetailsView(_ detailsView: SettingsDetailsView, didChangeTo state: SettingsDetailsView.State, with model: SettingsDetailsViewModel)
+    func settingsDetailsView(_ detailsView: SettingsDetailsView, didTapPrimaryButtonWith model: SettingsDetailsViewModel)
+}
+
+// MARK: - View
+public class SettingsDetailsView: UIView {
+
+    public enum State {
+        case normal
+        case details
+    }
+
+    // MARK: - Public properties
+
+    public weak var delegate: SettingsDetailsViewDelegate?
+
+    // MARK: - Private properties
+
+    private var state: State = .normal
+    private var model: SettingsDetailsViewModel?
+
+    // MARK: - Subviews
+
+    private lazy var iconView: UIImageView = {
+        let imageView = UIImageView(withAutoLayout: true)
+        return imageView
+    }()
+
+    private lazy var titleLabel: Label = {
+        let label = Label(style: .title3Strong, withAutoLayout: true)
+        label.textColor = .textPrimary
+        label.textAlignment = .center
+        label.numberOfLines = 0
+        return label
+    }()
+
+    private lazy var textLabel: Label = {
+        let label = Label(style: .bodyRegular, withAutoLayout: true)
+        label.textColor = .textPrimary
+        label.textAlignment = .center
+        label.numberOfLines = 0
+        return label
+    }()
+
+    private lazy var secondaryButton: Button = {
+        let button = Button(style: .flat, withAutoLayout: true)
+        button.addTarget(self, action: #selector(secondaryButtonTapped), for: .touchUpInside)
+        return button
+    }()
+
+    private lazy var primaryButton: Button = {
+        let button = Button(style: .callToAction, withAutoLayout: true)
+        button.addTarget(self, action: #selector(primaryButtonTapped), for: .touchUpInside)
+        return button
+    }()
+
+    private lazy var shadowView = DynamicShadowView(
+        withAutoLayout: true
+    )
+
+    private lazy var scrollView: UIScrollView = {
+        let scrollView = UIScrollView(withAutoLayout: true)
+        scrollView.delegate = self
+
+        scrollView.contentInset = UIEdgeInsets(
+            top: 0,
+            left: 0,
+            bottom: scrollViewBottomInset,
```

Comment:
i think you can just define `UIEdgeInsets(bottom: scrollViewBottomInset)`

Reply:
I'll fix